src/power_grid_model/validation/_validation.py

```diff
@@ -962,6 +982,63 @@ def validate_generic_power_sensor(data: SingleDataset, component: ComponentType)
     return errors
 
 
+def validate_generic_current_sensor(data: SingleDataset, component: ComponentType) -> list[ValidationError]:
+
+    errors = validate_base(data, component)
+    errors += _all_greater_than_zero(data, component, "i_sigma")
+    errors += _all_greater_than_zero(data, component, "i_angle_sigma")
+    errors += _all_valid_enum_values(data, component, "measured_terminal_type", MeasuredTerminalType)
```

Comment:
We would need a separate error InvalidEnumSubsetError? because we dont have current sensor on all terminal types.